packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1835,12 +1871,44 @@ def _handle_unhealthy_tm(self) -> Generator:
             not_ok_code = 1
             sys.exit(not_ok_code)
 
-        reset_successfully = yield from self.reset_tendermint_with_wait()
-        if not reset_successfully:
-            self.context.logger.error("Failed to reset tendermint.")
-            return
+    def _handle_unhealthy_tm(self) -> Generator:
+        """This method handles the case when the tendermint node is unhealthy."""
+        self.context.logger.warning(
+            "The local deadline for the next `begin_block` request from the Tendermint node has expired! "
+            "Trying to reset local Tendermint node as there could be something wrong with the communication."
+        )
 
-        self.context.logger.info("Tendermint reset was successfully performed. ")
+        # we first check whether the reason why we haven't received blocks for more than we allow is because
+        # there are not enough peers in the network to reach majority.
+        yield from self._kill_if_no_majority_peers()
+
+        # since we have reached this point that means that
```

Comment:
Incomplete docstring?

Reply:
Yes 😓. I have a WIP on my local that has some overlap with this. I can address it there if thats okay. It could spare us a CI run.